src/zon.zig

```diff
@@ -0,0 +1,831 @@
+
+    const fields: std.meta.fieldInfo(Zoir.Node, .struct_literal).type = switch (node.get(self.file.zoir.?)) {
+        .struct_literal => |fields| fields,
+        .empty_literal => .{ .names = &.{}, .vals = .{ .start = node, .len = 0 } },
+        else => return self.fail(
+            .{ .node_abs = node.getAstNode(self.file.zoir.?) },
+            "expected type '{}'",
+            .{res_ty.fmt(self.sema.pt)},
+        ),
+    };
+
+    const field_defaults = struct_info.field_inits.get(ip);
+    const field_values = try self.sema.arena.alloc(InternPool.Index, struct_info.field_names.len);
+    for (field_values) |*v| v.* = .none;
+
+    for (0..fields.names.len) |i| {
+        const field_name = try ip.getOrPutString(
+            gpa,
+            self.sema.pt.tid,
+            fields.names[i].get(self.file.zoir.?),
+            .no_embedded_nulls,
+        );
+        const field_node = fields.vals.at(@intCast(i));
+
+        const name_index = struct_info.nameIndex(ip, field_name) orelse {
+            return self.fail(
+                .{ .node_abs = field_node.getAstNode(self.file.zoir.?) },
+                "unexpected field '{}'",
+                .{field_name.fmt(ip)},
+            );
+        };
+
+        const field_type = Type.fromInterned(struct_info.field_types.get(ip)[name_index]);
+        if (field_values[name_index] != .none) {
+            const field_node_ast = field_node.getAstNode(self.file.zoir.?);
+            const field_name_token = self.file.tree.firstToken(field_node_ast) - 2;
+            return self.fail(
+                .{ .token_abs = field_name_token },
+                "duplicate field '{}'",
+                .{field_name.fmt(ip)},
+            );
+        }
+
+        field_values[name_index] = try self.lowerExpr(field_node, field_type);
+
+        if (struct_info.comptime_bits.getBit(ip, name_index)) {
+            const val = ip.indexToKey(field_values[name_index]);
+            const default = ip.indexToKey(field_defaults[name_index]);
+            if (!val.eql(default, ip)) {
+                const field_node_ast = field_node.getAstNode(self.file.zoir.?);
+                const field_name_token = self.file.tree.firstToken(field_node_ast) - 2;
+                return self.fail(
+                    .{ .token_abs = field_name_token },
+                    "value stored in comptime field does not match the default value of the field",
+                    .{},
+                );
+            }
+        }
+    }
+
+    for (0..field_values.len) |i| {
+        if (field_values[i] == .none and i < field_defaults.len) {
+            field_values[i] = field_defaults[i];
+        }
+    }
+
+    const field_names = struct_info.field_names.get(ip);
+    for (field_values, field_names) |*value, name| {
+        if (value.* == .none) return self.fail(
+            .{ .node_abs = node.getAstNode(self.file.zoir.?) },
+            "missing field {}",
```

Comment:
```suggestion
            "missing field '{}'",
```
(I just checked and the corresponding Sema error doesn't quote this name, which I guess is where you got this from; that's wrong, it should!)